[CLEANUP beta] #16391 Cleaning up the test output

KamiKillertO opened the pull request:
This resolves some Quest Tasks of Issue #16391:

- `Error: Catch me`
- `Error: the error.` 
- `Error: Element .does-not-exist not found.`

I've just remove a call to `console.error`, I don't know if he was here on purpose 🤷‍♂️.

KamiKillertO committed:
[CLEANUP beta] #16391 Cleaning up the test output

Remove console error `Error: Element .does-not-exist not found.`

rwjblue commented on packages/ember-testing/lib/test/adapter.js:
```diff
@@ -28,6 +28,4 @@ export function asyncEnd() {
 function adapterDispatch(error) {
   adapter.exception(error);
-  
-  console.error(error.stack);// eslint-disable-line no-console 
```
Yep, we need to bring this back (and explicitly mock in tests where we expect an error).

Specifically, this console.error is the _only_ thing that notifies the console of failures (including the stack trace) when unexpected exceptions or rejections happen in tests...

KamiKillertO replied:
Something like this?

```javascript
function adapterDispatch(error) {
   adapter.exception(error);
   
   adapter.outputError(error.stack); 
```

`outputError` is by default `console.error` and can be overridden with a noop like this.

```
Test.adapter = QUnitAdapter.create({
   outputError () {}
});
```

rwjblue replied:
I was more thinking to just override `console.error` in the specific tests, then reset it in the `afterEach`

KamiKillertO committed:
[CLEANUP beta] Override console.error in tests

Bring back console.error in `ember-testing/lib/test/adapter`. Instead override `console.error` with a noop function in tests

KamiKillertO committed:
[CLEANUP beta] Fix tests by adding missing call to super.afterEach

rwjblue commented:
Thank you!!
